### PR TITLE
Implement `RefCell::replace` and `RefCell::swap`

### PR DESCRIPTION
**Pull request by notriddle**
Tracking issue: #43570

### CODE REVIEW

File: src/libcore/cell.rs [hunk 1]
```diff
@@ -543,6 +543,59 @@ impl<T> RefCell<T> {
         debug_assert!(self.borrow.get() == UNUSED);
         unsafe { self.value.into_inner() }
     }
+
+    /// Replaces the wrapped value with a new one, returning the old value,
+    /// without deinitializing either one.
+    ///
+    /// This function corresponds to [`std::mem::replace`](../mem/fn.replace.html).
+    ///
+    /// # Example
```
Comment (frewsxcv) [inline src/libcore/cell.rs]: `s/Example/Examples`

https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md#using-markdown

File: src/libcore/cell.rs [hunk 2]
```diff
@@ -543,6 +543,59 @@ impl<T> RefCell<T> {
...
+    pub fn replace(&self, t: T) -> T {
+        mem::replace(&mut *self.borrow_mut(), t)
+    }
+
+    /// Swaps the wrapped value of `self` with the wrapped value of `other`,
+    /// without deinitializing either one.
+    ///
+    /// This function corresponds to [`std::mem::swap`](../mem/fn.swap.html).
+    ///
+    /// # Example
```
Comment (frewsxcv) [inline src/libcore/cell.rs]: `s/Example/Examples`

**Comment by notriddle** [comment]
Those problems are fixed now.

File: src/libcore/cell.rs [hunk 3]
```diff
@@ -543,6 +543,59 @@ impl<T> RefCell<T> {
...
+    /// let c = RefCell::new(5);
+    /// let d = RefCell::new(6);
+    /// c.swap(&d);
+    /// assert_eq!(c, RefCell::new(6));
+    /// assert_eq!(d, RefCell::new(5));
+    /// ```
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if the `RefCell` has any outstanding borrows,
```
Comment (sfackler) [inline src/libcore/cell.rs]: "either `RefCell`", right?

File: src/libcore/cell.rs [hunk 4]
```diff
@@ -543,6 +543,59 @@ impl<T> RefCell<T> {
         debug_assert!(self.borrow.get() == UNUSED);
         unsafe { self.value.into_inner() }
     }
+
+    /// Replaces the wrapped value with a new one, returning the old value,
```
Comment (sfackler) [inline src/libcore/cell.rs]: I wonder if the docs should mention that these are simply convenience functions for their one-liner implementations? I wouldn't want people thinking that they're more "magical" than they really are. Thoughts @steveklabnik? Not sure if we do something similar elsewhere.

File: src/libcore/cell.rs (hunk 3 above)
Comment (notriddle) [inline src/libcore/cell.rs]: There is only one `RefCell` involved in a `RefCell::replace`. This is different from `RefCell::swap`, which does actually involve two of them.

Comment (sfackler) [inline src/libcore/cell.rs]: Oops, yeah, I thought this was below.

**Comment by carols10cents** [comment]
ping @sfackler, looks like this is ready for your rereview!

**Pull request merged**